lib/foreman_puppet/engine.rb

```diff
@@ -33,7 +33,7 @@ def argument(name, type, *rest, loads: nil, **kwargs, &block)
     engine_name 'foreman_puppet'
     isolate_namespace ForemanPuppet
 
-    config.paths['db/migrate'] << 'db/migrate_foreman' if Gem::Dependency.new('', ">= #{ForemanPuppet::FOREMAN_EXTRACTION_VERSION}").match?('', SETTINGS[:version])
+    config.paths['db/migrate'] << 'db/migrate_foreman' if Gem::Dependency.new('', ">= #{ForemanPuppet::FOREMAN_DROP_MIGRATIONS_VERSION}").match?('', SETTINGS[:version])
```

Comment:
In theory, since you are saying `>=`, you are wanting this inclusion of migrations to happen with 3.1 which is effectively current nightly. However, we are not seeing this execute on nightly and we would face the problem where this does not execute until the actual release given:

```
irb(main):001:0> Gem::Dependency.new('', ">= 3.0").match?('', '3.0.0.rc2')
=> false
irb(main):002:0> Gem::Dependency.new('', ">= 3.0").match?('', '3.1.0-develop')
=> false
irb(main):003:0> Gem::Dependency.new('', ">= 3.0").match?('', '3.0.0')
=> true
```

So yea, this would be a stop gap fix for 3.0, but it would still result in incorrect behavior IMO. I am OK with fixing something to get 3.0 happily out the door, but it seems like maybe you should just remove this line all together for the 3.0 release?

Reply:
we can get this in and remove this line in `foreman_puppet 1.0`

Reply:
I've added notag and opened a https://github.com/theforeman/foreman_puppet/pull/186 for version 1.0